### PR TITLE
Fix broadcast_arrays return type

### LINKED ISSUE #544: cupy.broadcast_arrays returns tuple, while numpy returns list
CuPy: current master (e51b311)
```
>>> cupy.broadcast_arrays(cupy.ones((1,)), cupy.ones((2)))
(array([ 1.,  1.]), array([ 1.,  1.]))
>>> numpy.broadcast_arrays(numpy.ones((1,)), numpy.ones((2)))
[array([ 1.,  1.]), array([ 1.,  1.])]
```

### PR DESCRIPTION
**Pull request by niboshi**
Fixed #544 

Please merge #543 first.

### CODE REVIEW

**Comment by takagi** [comment]
Please resolve the conflicts.

**Comment by niboshi** [comment]
Rebased.

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2266,7 +2266,7 @@ cdef class broadcast:
                 a_sh = a._shape[a_ndim - i - 1]
                 if a_sh == r_shape[i]:
                     r_strides[i] = a._strides[a_ndim - i - 1]
-                elif a_sh != 1:
+                elif a_sh > 1:
```
Comment (takagi) [inline cupy/core/core.pyx]: This line seems to be left unchanged in #543.

File: tests/cupy_tests/manipulation_tests/test_dims.py [hunk 2]
```diff
@@ -268,6 +258,12 @@ def test_broadcast(self):
         self.assertEqual(broadcast_np.size, broadcast_cp.size)
         self.assertEqual(broadcast_np.nd, broadcast_cp.nd)
 
+    @testing.numpy_cupy_array_list_equal()
+    def test_broadcast_arrays(self, xp):
```
Comment (takagi) [inline tests/cupy_tests/manipulation_tests/test_dims.py]: Would you perform this test for all dtypes because the change seems to have degradation?

**Comment by niboshi** [comment]
Thank you for review!
Fixed those points.

**Comment by takagi** [comment]
I note that I don't make the v2 backport of this PR because this change is no-compat.

**Pull request merged**